File: lab5/DictionaryTest.c
//DictionaryTest.c
//Aaron Hom
//awhom
//lab5
//A client to test different functions from Dictionary.c

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"
#define MAX_LEN 180

int main(int argc, char* argv[]){  

  Dictionary library = newDictionary();
  char* x;
  char* y;
  char* book1[] = {"7", "11", "23", "27", "96"};
  char* book2[] = {"hello", "hola", "ni hao", "shalom", "czesc"};
  int i;

  for(i = 0; i < 5; i++){
    insert(library, book1[i], book2[i]);
  }

  printDictionary(stdout, library);

  printf("%d\n", size(library));
  printf("%s\n", (isEmpty(library)?"true":"false"));

  delete(library, "11");
  delete(library, "96");
  //delete(library, "12"); //returns non-existent key error

  //insert(library, "27", "hi") //returns duplicate key error

  printDictionary(stdout, library);
  printf("%d\n", size(library));

  makeEmpty(library);
  printf("%s\n", (isEmpty(library)?"true":"false"));

  freeDictionary(&library);
  return(EXIT_SUCCESS);
}

File: prog3/DictionaryTest.java
//DictionaryTest.java
//Aaron Hom
//awhom
//Program 3
//Serves as test client for Dictionary ADT

public class DictionaryTest{
    public static void main(String[] args){
        Dictionary book = new Dictionary();
	System.out.println(book.isEmpty());
	System.out.println(book.size());
 	book.insert("4", "hello");
	System.out.println(book.isEmpty());
	System.out.println(book.size());
	book.insert("2", "hola");
	book.insert("5", "hallo");
	System.out.println(book.isEmpty());
        System.out.println(book.size());
	book.delete("5");
	System.out.println(book.size());
	//book.insert("2", "ni hao"); //throws DuplicateKeyException
	//book.delete("7"); //throws NoKeyFoundException
	book.insert("3", "czesc");
        book.insert("1", "shalom");
	System.out.println(book.size());
	System.out.println(book);
	System.out.println(book.toString());
	book.makeEmpty();
	System.out.println(book.isEmpty());
	System.out.println(book.size());

    }
}

